hal/ia32/cpu.c

```diff
@@ -392,22 +418,36 @@ void *_cpu_initCore(void)
 {
 	cpu.ncpus++;
 
-	u32 a = *(u32 *)0xfee000f0;
-	*(u32 *)0xfee000f0 = (a | 0x100);
+	const unsigned int id = hal_cpuGetID();
```

Comment:
Let's keep with kernel convention of variable declaration strictly on the beginning of the function